fabric-commands-v0/src/main/java/net/fabricmc/fabric/api/registry/CommandRegistry.java

```diff
@@ -38,8 +39,9 @@
 	 * @param consumer  The command provider, consuming {@link CommandDispatcher}.
 	 */
 	public void register(boolean dedicated, Consumer<CommandDispatcher<ServerCommandSource>> consumer) {
-		CommandRegistrationCallback.EVENT.register(((dispatcher, isDedicated) -> {
-			if (dedicated && !isDedicated) {
+		CommandRegistrationCallback.EVENT.register(((dispatcher, registryAccess, environment) -> {
+			if (dedicated && environment == RegistrationEnvironment.INTEGRATED
+					|| !dedicated && environment == RegistrationEnvironment.DEDICATED) {
```

Comment:
Check `environment.dedicated` so that `RegistrationEnvironment.ALL` is handled properly.